src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -84,10 +89,13 @@ private void UpdateParameterMetadata(ModelMetadata metadata, IBinderMetadata bin
                 ValueProvider = bindingContext.ValueProvider,
             };
 
+            var modelState = actionContext.ModelState;
+            modelState.MaxAllowedErrors = _options.MaxModelValidationErrors;
```

Comment:
This change is being made for the opposite reason. The goal is to set `MaxModelValidationErrors` where model binding is called.